packages/core/src/event-listeners/index.ts

```diff
@@ -21,6 +22,9 @@ export const addOidcEventListeners = (provider: Provider, queries: Queries) => {
   provider.addListener('access_token.saved', async (token) => {
     return accessTokenIssuedListener(token, queries);
   });
+  provider.addListener('client_credentials.saved', async () => tokenUsageCounter(queries));
+  provider.addListener('client_credentials.issued', async () => tokenUsageCounter(queries));
+  provider.addListener('refresh_token.saved', async () => tokenUsageCounter(queries));
```

Comment:
nit: this may cause duplicate counts, maybe we can align one type of token to record

Reply:
Per offline discussion, count refresh token for now.